locust_tasks/locustfile.py

```diff
@@ -166,49 +189,163 @@ def correct_address(self):
-        self.client.post("/request-access-code/confirm-address", {
-            'request-address-confirmation': 'yes'
-        })
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Text message")
 
     @task(5)
+    def select_method(self):
+        """
+        POST 'sms' to select text message as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'sms'
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-SelectMethod', self, response, 200, Page.ENTER_MOBILE)
+
+    @task(6)
     def enter_mobile_number(self):
-        self.client.post("/request-access-code/enter-mobile", {
+        """
+        POST a mobile number
+        """
+        with self.client.post("/en/requests/access-code/enter-mobile/", {
             'request-mobile-number': '07714 330 933'
-        })
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-EnterMobileNumber', self, response, 200, Page.CONFIRM_MOBILE, "933")
 
-    @task(6)
+    @task(7)
     def confirm_mobile_number(self):
-        self.client.post("/request-access-code/confirm-mobile", {
+        """
+        POST 'yes' to confirm mobile number
+        """
+        with self.client.post("/en/requests/access-code/confirm-mobile/", {
             'request-mobile-confirmation': 'yes'
-        })
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-ConfirmMobileNumber', self, response, 200, Page.CODE_SENT)
+
+
+class RequestNewCodePost(SequentialTaskSet):
+
+    # All users arrive at the start page
+    @task(1)
+    def start_page(self):
+        """
+        GET Start page
+        """
+        self.case = get_next_case()
+        with self.client.get('/en/start/', catch_response=True) as response:
+            verify_response('RequestUacPost-Start', self, response, 200, Page.START)
+
+    @task(2)
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacPost-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
+
+    @task(3)
+    def select_address(self):
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        logger.info("Address selected: " + self.address_to_select)
```

Comment:
The Locust code can't do info text (except during development). 
It has to run cleanly unless it's to report an error

Reply:
I see. In that case shall I just comment it out so that it can be commented back in when doing development (I find it quite useful to know what address it's looking at).

Reply:
I've deleted this now